runner/engine.go

```diff
@@ -536,6 +537,7 @@ func (e *Engine) runBin() error {
 			case <-killCh:
 				return
 			default:
+				e.procKillWg.Add(1)
```

Comment:
What I don't understand is why the WaitGroup passed to `killFunc` on line 553 doesn't seem to be working properly, I'd expect that to have waited for the process to be killed.

Reply:
because killFunc is running in a goroutine

Reply:
No that's expected of course, but the WaitGroup is expects as an argument is defined on 522 and is only ever `Wait`ed on line 528, so perhaps waiting on that wasn't working as expected or the `Wait` wasn't being called every time. To me, at least, figuring that out seems like a better solution than adding yet another WaitGroup

Reply:
when i remove the wg.Done(), the we still can exit. i think it's pointer issue. because we put the pointer into funtion